src/commands.rs

```diff
@@ -566,6 +566,13 @@ struct TouchupArgs {
     /// The revision to touch up
     #[arg(long, short, default_value = "@")]
     revision: RevisionArg,
+    /// Revision to show on the left side of the diff
+    ///
+    /// Instead of showing the changes between the revision being edited and
+    /// its parents, show the diff between the revision being edited and the
+    /// revision specified here.
+    #[arg(long, value_name = "REVISION")]
+    diff_from: Option<RevisionArg>,
```

Comment:
We should probably replace the arguments by `--from` and `--to` and match `jj diff`'s behavior. This command is very similar to `jj diff`, and can even be used for viewing a diff in a GUI, as you pointed out. See `DiffArgs` for how we tell Clap which arguments are incompatible. Agree?

Reply:
I don't know if it makes sense to have separate `-r` and `--to`. With `--to`, it's less clear that the `--to` commit will be modified if the user makes any edits to the diff. Perhaps it would make sense to edit the *left* side of the commit if `--to` is specified, but not `--from`, but that's both complicated to implement and potentially confusing.

Reply:
I kind of see what you mean about `jj diffedit --to foo`, but I also feel like I would expect that to edit the `--to` side rather than the working copy. Btw, I have tried to avoid having one argument be called the generic `-r/--revision` when there are revisions that play different roles involved (like `--diff-from` and `-r`).

I also wonder if we should give `jj restore` an `-r` argument and make these commands aliases. It seems like they will do exactly the same thing, just described differently. Thanks for making me see the similarity between `touchup` and `restore`.

Reply:
I'll resolve this conversation and move it to the main body.